shared/api/auth.go

```diff
@@ -24,3 +24,140 @@ const (
 	// IdentityTypeOIDCClient represents an identity that authenticates with OIDC..
 	IdentityTypeOIDCClient = "OIDC client"
 )
+
+// Identity is the type for an authenticated party that can make requests to the HTTPS API.
```

Comment:
New structs in the API package should have `// API extension: ` line indicating when they got added.